odoo/tools/date_utils.py

```diff
@@ -199,6 +199,35 @@ def subtract(value, *args, **kwargs):
     """
     return value - relativedelta(*args, **kwargs)
 
+def ceil(value, granularity):
+    """
+    Return the ``value`` rounded up to a ``granularity``.
+
+    :param value: initial date or datetime.
+    :param granularity: year, month, day, hour, minute or second
+    :return: the resulting date/datetime.
+    """
+    granularities = (
+        ['year', 'month', 'day', 'hour', 'minute', 'second', 'microsecond']
+        if isinstance(value, datetime) else
+        ['year', 'month', 'day']
+    )
+    index = granularities.index(granularity, 0, len(granularities) - 1)
+    minvalues = {'month': 1, 'day': 1}
```

Comment:
These should be module-level constants, there's no good reason for them to be manipulated at runtime. And the valid granularities for `date` should probably be checked explicitly, if we really want to cater to this case at all.